### PR TITLE
Feature/favorites table queries ect

### LINKED ISSUE #3: Incorporate queries into fetch calls

### PR DESCRIPTION
**Pull request by ectweitmann**
## What Features are Being Added?
- Created multiple queries for the favorites table
  - POST a new book to favorites
  - DELETE a book from favorites
  - GET all books in favorites
- Added new `db` directory
- Added timestamp column to books table

## What is Being Refactored?
- Refactored `knexfile.js`, specifically the file paths for the migrations and seeds.
- Refactored server routes to utilized newly created queries. 

## Where should the reviewer start?
- `queries.js`, `server.js`, and the new migration file. 
## How should this be manually tested?
- Run the server and try to GET POST and DELETE from the favorites table in Postman

## What are the relevant tickets?
- Create queries for data retrieval and incorporate queries into fetch calls.

## Any Other Notes/Questions/Problems?
- What are the `select` and `from` methods doing in our `removeBookFromFavs` query/why are they necessary?

### CODE REVIEW

File: db/migrations/20220220181650_add_timestamp.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = function(knex) {
+  return knex.schema.table('books', (table) => {
+    table.timestamps(true, true);
```
Comment (cagallo) [inline db/migrations/20220220181650_add_timestamp.js:7]: Nice add!

File: db/migrations/20220220181650_add_timestamp.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/**
+ * @param { import("knex").Knex } knex
```
Comment (krdurrett) [inline db/migrations/20220220181650_add_timestamp.js:2]: Good idea to add timestamp to books table, should also probably add to favorites

File: db/queries.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const connection = require('../knexfile')['development'];
+const database = require('knex')(connection);
+
+module.exports = {
```
Comment (krdurrett) [inline db/queries.js:4]: These queries look great!

File: db/queries.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    getSingleBook(request) {
+      return database('books').where('id', request.params.id).select();
+    },
+    addBookToFavs(favorite) {
+      return database('favorites').insert(favorite, 'id');
+    },
+    removeBookFromFavs(request) {
+      return database('favorites').where('isbn', request.params.isbn)
+        .del()
+        .then(() => {
```
Comment (cagallo) [inline db/queries.js:17]: The .then that held us up forever! We should ask Robbie more about why this was needed when we have our standup with him.

File: db/queries.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+          return database.select().from('favorites');
```
Comment (Fordo29) [inline db/queries.js:18]: This was an interesting learning - I assume this is returning the favorites database after removing the one selected.

File: server.js [hunk 6]
```diff
@@ -17,55 +17,43 @@ app.get('/', (request, response) => {
...
+  queries.getAllFavs()
+    .then(data => response.status(200).json(data))
+    .catch(error => response.status(500).json({ error }))
 });
 
 app.post('/api/v1/favorites', (request, response) => {
   const favorite = request.body;
-  const { id, title, description, amazonLink, author, bookImage } = favorite;
-  for (let requiredParameter of ['id', 'title', 'description', 'amazonLink', 'author', 'bookImage']) {
+  const { title, description, 'amazon_link': amazonLink, author, 'book_image': bookImage } = favorite;
```
Comment (cagallo) [inline server.js:39]: Great memory with remembering how to do this from Rancid Tomatillos!

File: server.js [hunk 7]
```diff
@@ -17,55 +17,43 @@ app.get('/', (request, response) => {
...
+  queries.addBookToFavs(favorite)
+    .then(data => response.status(201).json(data))
+    .catch(error => response.status(500).json({ error }));
+});
 
+app.delete('/api/v1/favorites/:isbn', (request, response) => {
+  queries.removeBookFromFavs(request);
   response.status(200).json({
-    message: `Book #${id} has been deleted from favorites`
+    message: `Book with isbn #${request.params.isbn} has been deleted from favorites`
```
Comment (cagallo) [inline server.js:54]: Smart to delete with isbn vs id!

File: server.js [hunk 8]
```diff
@@ -17,55 +17,43 @@ app.get('/', (request, response) => {
...
-    })
-  }
-
-  app.locals.favorites = favorites.filter(fav => fav.id !== id)
+  queries.addBookToFavs(favorite)
+    .then(data => response.status(201).json(data))
+    .catch(error => response.status(500).json({ error }));
+});
 
+app.delete('/api/v1/favorites/:isbn', (request, response) => {
```
Comment (krdurrett) [inline server.js:51]: Great job with these functions. I believe we can add in some more error handling

**Review by cagallo — APPROVED** [review]

File: server.js (hunk 6 above)
Comment (Fordo29) [inline server.js:39]: nice find on adding snake case

**Review by krdurrett — APPROVED** [review]
So much work! Great job with all of the added features in the db

**Pull request merged**